File: eat-it/src/main/java/ide/eatit/model/responses/ItemsResponse.java
package ide.eatit.model.responses;

import java.util.List;

public class ItemsResponse<T> extends BaseResponse {

    private List<T> items;

    public ItemsResponse(String statusCode, String message, String status, List<T> items) {
        super(statusCode, message, status);
        this.items = items;
    }

    public List<T> getItems() {
        return items;
    }

    public void setItems(List<T> items) {
        this.items = items;
    }
}

File: eat-it-fe/src/app/models/responses/items-response.ts
import { BaseResponse } from "./base-response";

export class ItemsResponse<T> extends BaseResponse {
  items: T[];

  constructor(statusCode: string, message: string, status: string, items: T[]) {
    super(statusCode, message, status);
    this.items = items;
  }

  static override failedResponse(message: string = ""): BaseResponse {
    return new BaseResponse('500', "BE response was unexpected." + message, 'FAILED');
  }
}
